mm2src/mm2_net/src/native_http.rs

```diff
@@ -1,12 +1,212 @@
-use crate::transport::{SlurpError, SlurpResult, SlurpResultJson};
-use common::wio::{drive03, HYPER};
-use common::APPLICATION_JSON;
+//! Facilitates execution of http requests
+//!
+//! # Layout
+//!
+//! This module contains several service functions like [slurp_post_json] or [slurp_url_with_headers]
+//! for executing http protocol requests, allowing you to set certain headers, make a [`Request`] in JSON or [`Body`] format.
+//!
+//! These methods are wrappers over [`HYPER`], which is actually `Client<HttpsConnector<HttpConnector>>` that implements
+//! [`SlurpHttpClient`] trait designed to provide http capabilities through it.
+//!
+//! There are also facilities for constructing [SlurpError] from the [hyper::Error]
+//!
+
+use async_trait::async_trait;
 use futures::channel::oneshot::Canceled;
 use http::{header, HeaderValue, Request};
-use hyper::Body;
-use mm2_err_handle::prelude::*;
+use hyper::client::connect::Connect;
+use hyper::client::ResponseFuture;
+use hyper::{Body, Client};
 use serde_json::Value as Json;
 
+use common::wio::{drive03, HYPER};
+use common::APPLICATION_JSON;
+use mm2_err_handle::prelude::*;
+
+use super::transport::{SlurpError, SlurpResult, SlurpResultJson};
+
+/// Provides requesting http through it
+///
+/// Initially designed to be used with [hyper::Client] that could be constructed in different specific ways.
+/// one of wich is using with statically defined [HYPER] that is common client able to request https or https urls
```

Comment:
typo in `wich`

Reply:
Thank you for review and sorry for long fixing this typo ) 
[done](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1910/commits/d254c2186083f4d22b1b582e1549ed75fd68612f#diff-b3f52bbaf2ac76298e4e9ad9b06a473c103b13c34858f8d26b9df92826bce5d2R31)